src/allmydata/test/test_checker.py

```diff
@@ -286,32 +395,232 @@ def test_check_and_repair(self):
                                              summary="worse",
                                              **data)
         crr.post_repair_results = post_cr
-        html = self.render2(w)
+        html = self.render_element(w)
         s = self.remove_tags(html)
 
         self.failUnlessIn("File Check-And-Repair Results for SI=2k6avp", s)
         self.failUnlessIn("Not Recoverable! : worse", s)
         self.failUnlessIn("Repair unsuccessful", s)
         self.failUnlessIn("Post-Repair Checker Results:", s)
 
-        d = self.render_json(w)
+        w2 = web_check_results.CheckAndRepairResultsRenderer(c, crr)
+        d = self.render_json(w2)
         def _got_json(data):
             j = json.loads(data)
             self.failUnlessEqual(j["repair-attempted"], True)
             self.failUnlessEqual(j["storage-index"],
                                  "2k6avpjga3dho3zsjo6nnkt7n4")
             self.failUnlessEqual(j["pre-repair-results"]["summary"], "illing")
             self.failUnlessEqual(j["post-repair-results"]["summary"], "worse")
-        d.addCallback(_got_json)
+        _got_json(d)
```

Comment:
Same as comment below (just get rid of the inline function since it's called synchronously here).

Reply:
Did this in ca6cf88.